etl/tasks.py

```diff
@@ -14,3 +17,24 @@ def import_file_task(
         is_background=True,
         is_remote=True,
     )
+
+
+@shared_task
+def remove_file_task(files: []):
+    account = None
+    for f in files:
+        remove = File.objects.filter(id=f).first()
+        account = remove.account
+        logger.info(
+            f"Recovering {remove.account.title} by removing failed file: {remove.filename}"
+        )
+        if remove:
+            remove.delete()
+
+    if account.files.count() == 0:
```

Comment:
It looks like there's a path where this will still get triggered when ``account = None`` and raise an exception. If the above comment is implemented, I'd suggest intending this to be within the for loop, like:

```python
f.delete()
if f.account.files.count() == 0:
    f.account.delete()
```